metaflow/tutorials/09-hellokfp/kfp_helloworld.py

```diff
@@ -0,0 +1,40 @@
+from kfp.components import func_to_container_op
+
+from metaflow import FlowSpec, step, kfp
+
+
+@func_to_container_op
+def my_hello_step_op_func(who: str) -> str:
+    message = f"hello {who}"
+    print(message)
+    return message
+
+
+class KfpHelloWorld(FlowSpec):
+    """
+    A Flow that decorates a Metaflow Step with a KFP component
+    """
+
+    @step
+    def start(self):
+        """
+        kfp.kfp_component_inputs Flow state ["who"] is passed to the KFP component as arguments
+        """
+        self.who = "world"
+        self.next(self.end)
+
+    @kfp(
+        container_op_func=my_hello_step_op_func,
+        kfp_component_inputs=["who"],
```

Comment:
I'm not fully clear that the variable `"who"` is referring to here. It could be the parameter `who` in the `my_hello_step_op_func` or variable `self.who` in the `start` step. I'd recommend changing one of these variable names to make that clearer.